### PR TITLE
OP-1341 Enable Medical Type for Soft Deletion

### PR DESCRIPTION
**Pull request by mohit10696**
See OP-1341.
Paired with https://github.com/informatici/openhospital-core/pull/1458.

### CODE REVIEW

**Comment by mohit10696** [comment]
image

image

File: src/main/java/org/isf/medtype/gui/MedicalTypeBrowserEdit.java [hunk 1]
```diff
@@ -263,6 +270,14 @@ private JTextField getCodeTextField() {
 		}
 		return codeTextField;
 	}
+	
+	private JCheckBox getActiveField() {
+		if (activeCheckbox == null) {
+			activeCheckbox = new JCheckBox("Active");
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medtype/gui/MedicalTypeBrowserEdit.java]: shouldn't this text be localized?

File: src/main/java/org/isf/medtype/gui/MedicalTypeBrowserEdit.java [hunk 2]
```diff
@@ -276,7 +291,9 @@ private JPanel getJDataPanel() {
 			jDataPanel.add(getCodeTextField());
 			jDataPanel.add(new JLabel(MessageBundle.getMessage("angal.common.description.txt") + ':'));
 			jDataPanel.add(getDescriptionTextField());
-			SpringUtilities.makeCompactGrid(jDataPanel, 2, 2, 5, 5, 5, 5);
+			jDataPanel.add(new JLabel(MessageBundle.getMessage("Active") + ':'));
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medtype/gui/MedicalTypeBrowserEdit.java]: "Active" isn't a typical message bundle string key; they start with `angal.`

File: src/main/java/org/isf/medtype/gui/MedicalTypeBrowserEdit.java (hunk 1 above)
Comment (mwithi) [reply src/main/java/org/isf/medtype/gui/MedicalTypeBrowserEdit.java]: I would not use "Active" but rather "Deleted" (so default is `false`) as in the rest of the application

**Comment by mohit10696** [comment]
@dbmalkovsky , Can someone of you confirm about the text ? 
Do you want as active over there or deleted over there? After that, I will localize that text.

**Comment by mwithi** [comment]
> Can someone of you confirm about the text ? Do you want as active over there or deleted over there? After that, I will localize that text.

Since we have already fields like XXX_ACTIVE in the DB (for future uses), I would use "Deleted" (like the new field) to avoid confusion.

**Comment by mohit10696** [comment]
@mwithi  Refactored active to deleted text and corresponding logic. Hoping to get  PR merged 🤞

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**